pkg/resource/resource.go

```diff
@@ -18,50 +18,42 @@ limitations under the License.
 package resource
 
 import (
-	"encoding/json"
-	"fmt"
-	"io"
+	"log"
 	"strings"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
-	"sigs.k8s.io/kustomize/pkg/gvk"
+	"sigs.k8s.io/kustomize/internal/k8sdeps"
 	"sigs.k8s.io/kustomize/pkg/ifc"
 	internal "sigs.k8s.io/kustomize/pkg/internal/error"
 	"sigs.k8s.io/kustomize/pkg/patch"
 	"sigs.k8s.io/kustomize/pkg/resid"
 )
 
-// Resource is an "Unstructured" (json/map form) Kubernetes API resource object
+// Resource is map representation of a Kubernetes API resource object
 // paired with a GenerationBehavior.
 type Resource struct {
-	unstructured.Unstructured
-	b ifc.GenerationBehavior
+	funStruct ifc.FunStruct
+	b         ifc.GenerationBehavior
 }
 
 // NewResourceWithBehavior returns a new instance of Resource.
 func NewResourceWithBehavior(obj runtime.Object, b ifc.GenerationBehavior) (*Resource, error) {
```

Comment:
FYI, `NewResourceWithBehavior` is called only when creating ConfigMaps and Secrets. Later we can cover them by interface, then this function can be removed from here.